packages/preact/src/hooks/useSpeechToText.ts

```diff
@@ -0,0 +1,110 @@
+import { useCallback, useEffect, useRef, useState } from "preact/hooks"
+
+interface SpeechToTextOptions {
+  language?: string
+  interimResults?: boolean
+}
+
+interface StartListeningOptions {
+  onResult: ({ value, confidence }: { value: string; confidence: number }) => void
+  onError?: (error: string) => void
+}
+
+/**
+ * Preact hook that provides speech-to-text functionality using the Web Speech API.
+ * @example
+ * ```jsx
+ * import { useNostoAppState, useSpeechToText } from "@nosto/search-js/preact/hooks"
+ *
+ * export default function MyComponent() {
+ *  const { newSearch } = useActions()
+ *  const { startListening, stopListening, isListening, isSupported } = useSpeechToText()
+ *  if (!isSupported) {
+ *    return null
+ *  }
+ *
+ *  return (
+ *    <button
+ *     onClick={() => {
+ *       if (isListening) {
+ *         stopListening()
+ *       } else {
+ *         startListening({
+ *           onResult: ({ value }) => {
+ *             newSearch({
+ *               query: value
+ *             })
+ *           }
+ *         })
+ *       }
+ *     }}
+ *    >
+ *     🎤︎︎
+ *    </button>
+ *  )
+ * }
+ * ```
+ */
+export function useSpeechToText(options?: SpeechToTextOptions) {
+  const { language = "en-US", interimResults = false } = options || {}
+
+  const [isListening, setIsListening] = useState(false)
+  const [isSupported, setIsSupported] = useState(true)
```

Comment:
drop is prefix

Reply:
do we need this to be managed via useState?
the supported state doesn't change and should be static